### PR TITLE
Optimize Spring samples

### PR DESCRIPTION
**Pull request by sdeleuze**
Hi, I am Sébastien one of the Spring Framework core committers. Here is a PR where @rstoyanchev and I contribute some optimization to `spring-webflux` and where we switch `spring` to use functional web end points like the WebFlux project for consistency and performance reasons.

### CODE REVIEW

File: frameworks/Java/spring/src/main/java/hello/web/JsonHandler.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.function.HandlerFunction;
+import org.springframework.web.servlet.function.ServerRequest;
+import org.springframework.web.servlet.function.ServerResponse;
+
+@Component
+public class JsonHandler implements HandlerFunction<ServerResponse> {
+
+	private static final Map<String, String> MESSAGE = Map.of("message", "Hello, world!");
```
Comment (NateBrady23) [inline frameworks/Java/spring/src/main/java/hello/web/JsonHandler.java:18]: Per the rules of this test, the object needs to be instantiated on each request. This optimization is not allowed.

Comment (sdeleuze) [reply frameworks/Java/spring/src/main/java/hello/web/JsonHandler.java:18]: Sorry I missed that, I will update the PR accordingly.

**Comment by sdeleuze** [comment]
@NateBrady23 I have updated both `spring` and `spring-webflux` projects to take in account your feedback and comply with the tests requirements.

**Comment by sebastienros** [comment]
Hey, I am Sébastien too, but from the ASP.NET team, let me tell you that this PR has made huge gains on the benchmarks (at least on our runs).

image

**Comment by sdeleuze** [comment]
Thanks for your feedback Sébastien ^^.

**Pull request merged**